tests/types/test_numpy.py

```diff
@@ -0,0 +1,376 @@
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (
+            np.ubyte(0),
+            "'0'::int2",
+        ),
+        (
+            np.ubyte(255),
+            "'255'::int2",
+        ),
+        # ubyte overflow
+        (
+            np.ubyte(-1),
+            "'255'::int2",
+        ),
+        (
+            np.ubyte(256),
+            "'0'::int2",
+        ),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_ubyte_(numpy_conn, val, expr, fmt_in):
+
+    assert isinstance(val, np.ubyte)
+    assert np.ubyte is np.uint8
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (
+            np.ushort(0),
+            "'0'::int4",
+        ),
+        (
+            np.ushort(65_535),
+            "'65535'::int4",
+        ),
+        # ushort overflow
+        (
+            np.ushort(-1),
+            "'65535'::int4",
+        ),
+        (
+            np.ushort(65_536),
+            "'0'::int4",
+        ),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_ushort_(numpy_conn, val, expr, fmt_in):
+
+    assert isinstance(val, np.ushort)
+    assert np.ushort is np.uint16
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (
+            np.uintc(0),
+            "'0'::int8",
+        ),
+        (
+            np.uintc(4_294_967_295),
+            "'4294967295'::int8",
+        ),
+        # uintc overflow
+        (
+            np.uintc(-1),
+            "'4294967295'::int8",
+        ),
+        (
+            np.uintc(4294967296),
+            "'0'::int8",
+        ),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_uintc_(numpy_conn, val, expr, fmt_in):
+
+    assert isinstance(val, np.uintc)
+    assert np.uintc is np.uint32
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (
+            np.uint(0),
+            "'0'::int2",
+        ),
+        (
+            np.uint(4_294_967_295),
+            "'4294967295'::int8",
+        ),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_uint(numpy_conn, val, expr, fmt_in):
+
+    assert isinstance(val, np.uint)
```

Comment:
I think that the problem you are having with the imports is caused by using np objects in the fixtures. You should use pure Python numbers.

```suggestion
        (
            4_294_967_295,
            "'4294967295'::int8",
        ),
    ],
)
@pytest.mark.parametrize("fmt_in", PyFormat)
def test_dump_numpy_uint(numpy_conn, val, expr, fmt_in):

    val = np.uint(val)
```